### PR TITLE
Initial pass on client

### PR DESCRIPTION
**Pull request by benogle**
- The client
- Tests (integration; starts a server)
- Docs in readme

### CODE REVIEW

File: src/stride.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  isValidEndpoint: /^(\/collect|\/process|\/analyze)/,
+}
+
+const ValidationMessages = {
+  hasLeadingSlash: 'Endpoints should have leading slash. e.g. `/collect`',
+  missingVersionNumber: 'Please omit the version number. e.g. `/collect`',
+  isValidEndpoint: 'Only the `/collect`, `/process`, and `/analyze` endpoints are supported',
+}
+
+const SupportedURLPatterns = {
```
Comment (benogle) [inline src/stride.js:153]: @usmanm, think this is too aggressive for validation? Basically it whitelists all the available endpoints.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,126 @@
 # stride.js
-JavaScript client for Stride
+
```
Comment (benogle) [inline README.md:2]: Take a peek at these and find awkwardness?

File: test/mock-server.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  },
+
+  '/v1/collect/error': function (req, res) {
+    let out = {message: 'bad request'}
+    res.writeHead(400, { 'Content-Type': 'application/json' })
+    res.end(`${JSON.stringify(out)}\n`)
+  },
+
+  '/v1/collect/success/subscribe': function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/plain' }) // FIXME: usmanm, is this type correct?
```
Comment (benogle) [inline test/mock-server.js]: Not sure if this is the correct content type. It's not strictly json cause no list object wraps the events

File: src/stride.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  POST: [
+    /^\/collect$/,
+    /^\/collect\/\w+$/,
+    /^\/process\/\w+$/,
+    /^\/analyze$/,
+    /^\/analyze\/\w+$/
+  ],
+  DELETE: [
+    /^\/collect\/\w+$/,
+    /^\/process\/\w+$/
```
Comment (usmanm) [inline src/stride.js]: Analyze also allows `DELETE`

File: src/stride.js (hunk 1 above)
Comment (usmanm) [reply src/stride.js:153]: This is good, in fact resource names can be validated even more aggressively than `w+`. See: https://github.com/pipelinedb/stride/blob/master/utils/validate.go#L5

File: test/mock-server.js (hunk 3 above)
Comment (usmanm) [reply test/mock-server.js]: Response header is always `Content-Type: application/json; charset=utf-8`

**Review by usmanm — APPROVED** [review]

**Pull request merged**